pkg/controllers/monitoring/monitoring-stack/components.go

```diff
@@ -214,7 +215,7 @@ func storageForPVC(pvc *corev1.PersistentVolumeClaimSpec) *monv1.StorageSpec {
 	}
 }
 
-func newRoleBinding(ms *stack.MonitoringStack, rbacResourceName string) *rbacv1.RoleBinding {
+func newRoleBindingForRoleType(ms *stack.MonitoringStack, rbacResourceName, roleType string) *rbacv1.RoleBinding {
```

Comment:
(suggestion) an alias type for role reference's kind would help readability and be more error-proof IMHO.

```suggestion
type roleRefKind string

const (
  clusterRoleKind = roleRefKind("ClusterRole")
  roleKind = roleRefKind("Role")
)

func newRoleBindingForRoleType(ms *stack.MonitoringStack, rbacResourceName string, kind roleRefKind) *rbacv1.RoleBinding {
```

Reply:
I did think of this but felt like an overkill :) but I agree it is better to be safe than sorry

Reply:
Since we are supporting namespace selector for Alertmanager, but Prometheus and Alertmanager roles are now Cluster roles.